### PR TITLE
rustfmt 1.x bump rustc-ap to v642

### LINKED ISSUE #3903: Bumping syntax-ap crates fails due to reduced visiblities
I'm trying to fix https://github.com/rust-osdev/uefi-rs/pull/104#issuecomment-548789808.  Rustfmt needs a new engough `syntax` to support `"efiapi"`. I tired to do what https://github.com/rust-lang/rustfmt/pull/3870 did and bump `rustc-ap-syntax` to version `626`.

This caused a very large number of compiler errors. Some of them are caused by small changes like: https://github.com/rust-lang/rust/commit/742ec4b9bf7ae7d693da7fe75e5f974e0fafb9d0, which are easy enough to fix. However, others errors are caused by changes like https://github.com/rust-lang/rust/commit/c189565edc5c9fc516170885b3a3061b936205fb that make a bunch of `libsyntax`'s functionality private. 

What should I do here? Should the `libsyntax` commits be reverted in the main rust-lang/rust repo?

CC: @Centril

### LINKED ISSUE #4009: `cargo fmt` fails with `half_open_range_patterns`
This example compiles fine with latest nightly, but `cargo fmt` fails. The failure also occurs on clippy's CI (https://travis-ci.com/rust-lang/rust-clippy/jobs/274511513).
Probably we need rustup and update submodule on rust-lang/rust?

```rust
#![feature(half_open_range_patterns)]
#![feature(exclusive_range_pattern)]

fn main() {
    match 54 {
        5.. => println!("yay!"),
        _ => (),
    }
}
```

error log:
```
error: `X..` range patterns are not supported
 --> main.rs:6:9
  |
6 |         5.. => println!("yay!"),
  |         ^^^ help: try using the maximum value for the type: `5..MAX`
```

### PR DESCRIPTION
**Pull request by calebcartwright**
Bumps the rustc-ap deps to v642 for the latest 1.x version of rustfmt 

Refs https://github.com/rust-lang/rust/issues/68917
Resolves #3903 
Resolves #4009

### CODE REVIEW

File: src/formatting.rs [hunk 1]
```diff
@@ -710,45 +715,57 @@ fn parse_crate(
 struct SilentOnIgnoredFilesEmitter {
     ignore_path_set: Rc<IgnorePathSet>,
     source_map: Rc<SourceMap>,
-    emitter: EmitterWriter,
+    emitter: Box<dyn Emitter + Send>,
     has_non_ignorable_parser_errors: bool,
     can_reset: Rc<RefCell<bool>>,
 }
 
+impl SilentOnIgnoredFilesEmitter {
```
Comment (calebcartwright) [inline src/formatting.rs]: The changes in e72f307f15138a8bb4fd1ae9ef3256fe1fa9dd94 on the emitters (and associated unit tests below) are a backport of a bug fix that's already been merged into master/2.x (https://github.com/rust-lang/rustfmt/pull/3920#pullrequestreview-317346826), but is missing from rustfmt 1.4.11.

This bug has been seen by a few users, including in the main rust-lang/rust repo (https://github.com/rust-lang/rustfmt/pull/3972#issuecomment-567566213) so I figured it'd be good to get this fix in along with the rustc-ap-* bump to get the bug fixed in rustfmt 1.x

Comment (topecongiro) [reply src/formatting.rs]: Thanks!

**Comment by topecongiro** [comment]
@calebcartwright Thank you for your work! I was about to ask whether you could make a pull request that backports the #4022 to the rustfmt-1.4.11 branch, but then I found out that you have already created the PR! Thanks a million.

I changed the base branch to rustfmt-1.4.12, which is based on rustfmt-1.4.11, to keep the branch in the correct state.

**Comment by calebcartwright** [comment]
@topecongiro when you get a chance can you publish v1.4.12 to crates.io?

I've gotten Racer bumped to rustc-ap-* v642 as well (https://github.com/racer-rust/racer/pull/1091), and IIUC once v1.4.12 is published I can open a PR to update Rls so that Rls and rustfmt can then be fixed and available again on nightlies.

**Pull request merged**